### DISCUSSION TITLE
stringgling to understant the ISAKMP id's behind NAT

### DISCUSSION THREAD

**Discussion by Giveback-ru** [strongswan/strongswan#1913] [Q&A]
Hello, 

I have one VM behind a NAT in the public cloud, I've setup up (successfully) a VTI tunnel using strongswan. It's NAT'ed public IP is 158.160.32.170, and the IP which the VM is aware of is 10.10.1.13. Since I've been always staying away from NATs and tunnels combination since it's just wrong, I'm struggling to understand which entity is which in strongswan logs:

````
Sep 19 10:59:58 web-ng charon-systemd[261069]: looking for peer configs matching 10.10.1.13[158.160.32.170]...83.222.71.53[83.222.71.53]
Sep 19 10:59:58 web-ng charon-systemd[261069]: no matching peer config found
````

This happens when the swanctl.conf part looks like:

````
connections {
    hq {
        local {
            id = 10.10.1.13
            auth = psk
        }

````

I've solved that by just changing the above to

````
connections {
    hq {
        local {
            id = %any
            auth = psk
        }
````

But since it matches just anything I feel it can be more precise. Could you help me to get which part does stand what for ? What it the part without brackets and what is the part in brackets ? Since 99.99% of my setups just has these identical and I've never really been thinking about these.

Should I have two connections may be ? One for case when hq initiates and one for case when this VM initiates an ISAKMP session ?

Thanks.

**Comment by tobiasbrunner** [comment]
See [this FAQ entry](https://docs.strongswan.org/docs/5.9/support/faq.html#_no_matching_peer_config_found).

**Comment by Giveback-ru** [comment]
Thanks, but this answers only half of the question. What do I set the id to so it matches the whole pair of IPs ? Still not getting it.

**Reply by tobiasbrunner** [reply]
It gives you the information you need to answer the other question yourself.
